v2/src/log_context_property_type_ascii_char_ptr.c

```diff
@@ -0,0 +1,135 @@
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_003: [ Otherwise, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string` shall copy the string pointed to by `property_value` to `buffer` by using `snprintf` with `buffer`, `buffer_length` and format string `%s` and pass in the values list the `const char*` value pointed to be `property_value`. ]*/
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_004: [ `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string` shall succeed and return the result of `snprintf`. ]*/
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_005: [ If any error is encountered, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string` shall fail and return a negative value. ]*/
+        result = snprintf(buffer, buffer_length, "%s", (const char*)property_value);
+    }
+    return result;
+}
+
+static int ascii_char_ptr_log_context_property_type_copy(void* dst_value, void* src_value)
+{
+    int result;
+
+    if (
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_007: [ If `dst_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).copy` shall fail and return a non-zero value. ]*/
+        (dst_value == NULL) ||
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_006: [ If `src_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).copy` shall fail and return a non-zero value. ]*/
+        (src_value == NULL)
+        )
+    {
+        (void)printf("Invalid arguments: void* dst_value=%p, void* src_value=%p\r\n",
+            dst_value, src_value);
+        result = MU_FAILURE;
+    }
+    else
+    {
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_008: [ `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).copy` shall copy the entire string (including the `NULL` terminator) from `src_value` to `dst_value`. ]*/
+        (void)strcpy(dst_value, src_value);
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_009: [ `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).copy` shall succeed and return 0. ]*/
+        result = 0;
+    }
+
+    return result;
+}
+
+static void ascii_char_ptr_log_context_property_type_free(void* value)
+{
+    /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_018: [ `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).free` shall return. ]*/
+    (void)value;
+}
+
+static LOG_CONTEXT_PROPERTY_TYPE ascii_char_ptr_log_context_property_type_get_type(void)
+{
+    /* Codes_SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_010: [ `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).get_type` shall return the property type `LOG_CONTEXT_PROPERTY_TYPE_ascii_char_ptr`. ]*/
+    return LOG_CONTEXT_PROPERTY_TYPE_ascii_char_ptr;
+}
+
+int LOG_CONTEXT_PROPERTY_TYPE_INIT(ascii_char_ptr)(void* dst_value, const char* format, ...)
+{
+    va_list args;
+    va_start(args, format);
```

Comment:
can be postponed until needed. (After parameter validation)

Reply:
Moved